app/services/provider_interface/hesa_data_export.rb

```diff
@@ -0,0 +1,145 @@
+
+      CSV.generate do |csv|
+        csv << csv_headings
+
+        applications.each do |application|
+          first_degree = application.application_form.application_qualifications
+            .order(created_at: :asc)
+            .find_by(level: 'degree')
+
+          csv << [
+            application.application_form.support_reference,
+            application.status,
+            application.application_form.first_name,
+            application.application_form.last_name,
+            application.application_form.date_of_birth,
+            application.application_form.first_nationality, # nationality
+            application.application_form.country, # domicile
+            application.application_form.candidate.email_address,
+            application.application_form.recruitment_cycle_year,
+            application.provider.code,
+            application.course.accredited_provider&.name,
+            application.course.code,
+            application.site.code,
+            study_mode(application),
+            subject_codes(application),
+            qualification_aim(application),
+            degrees_completed(application),
+            pad_hesa_value(first_degree, :qualification_type_hesa_code, 3),
+            pad_hesa_value(first_degree, :subject_hesa_code, 4),
+            pad_hesa_value(first_degree, :grade_hesa_code, 2),
+            first_degree&.institution_country,
+            first_degree&.start_year,
+            first_degree&.award_year,
+            pad_hesa_value(first_degree, :institution_hesa_code, 4),
+          ] + diversity_information(application)
+        end
+      end
+    end
+
+  private
+
+    def pad_hesa_value(degree, method, pad_by)
+      return 'no degree' if degree.blank?
+
+      code = degree.send(method)
+      return 'no data' if code.blank?
+
+      code.to_s.rjust(pad_by, '0')
+    end
+
+    def application_statuses_for(statuses)
+      result = []
+      statuses.each do |status|
+        result << STATUSES[status]
+      end
+      result.flatten.compact.uniq
+    end
+
+    def diversity_information(application)
+      return ['no data', 'no data', 'no data'] if application.application_form.equality_and_diversity.blank?
+
+      [
+        application.application_form.equality_and_diversity['hesa_sex'] || 'not specified',
+        (application.application_form.equality_and_diversity['hesa_disabilities'] || ['not specified']).join(' '),
+        application.application_form.equality_and_diversity['hesa_ethnicity'] || 'not specified',
+      ]
+    end
+
+    def csv_headings
+      [
+        'id', 'status', 'first name', 'last name', 'date of birth', 'nationality', 'domicile', 'email address',
+        'recruitment cycle', 'provider code', 'accredited body', 'course code', 'site code', 'study mode', 'SBJCA',
+        'QLAIM', 'FIRSTDEG', 'DEGTYPE', 'DEGSBJ', 'DEGCLSS', 'institution country', 'DEGSTDT', 'DEGENDDT',
+        'institution details', 'sex', 'disabilities', 'ethnicity'
+      ]
```

Comment:
This is not unlikely to go out of sync with the data array. Can we generate a hash instead like some of the other exports?

https://github.com/DFE-Digital/apply-for-teacher-training/blob/da381583a804669e7651849cfe4d54d3988af557/app/services/support_interface/application_choices_export.rb#L6-L23

You can use it with SafeCSV like this:

https://github.com/DFE-Digital/apply-for-teacher-training/blob/168da216862bfdb4a588d683d2358cb77a5a2a2c/app/workers/data_exporter.rb#L19-L22

Reply:
@tijmenb :+1: https://github.com/DFE-Digital/apply-for-teacher-training/pull/3135/commits/88021a8b53b528ac2655785432a5852faa28c012